File: python/1 - Iniciante/1047_tempo_de_jogo_com_minutos.py
hora_inicial, minuto_inicial, hora_final, minuto_final = map(int, input().split())

if hora_inicial < hora_final or (hora_inicial == hora_final and minuto_inicial < minuto_final):
    horas = hora_final - hora_inicial
    minutos = minuto_final - minuto_inicial

elif hora_inicial == hora_final and minuto_inicial == minuto_final:
    horas = 24
    minutos = 0

else:
    horas = (24 - hora_inicial) + hora_final
    minutos = minuto_final - minuto_inicial

if minutos < 0:
    minutos += 60
    horas -= 1

print(f"O JOGO DUROU {horas} HORA(S) E {minutos} MINUTO(S)")

File: c/c99/1 - Iniciante/1047_tempo_de_jogo_com_minutos.c
#include <stdio.h>

int main() {
    int h_inicio, m_inicio, h_final, m_final, horas, minutos;
    scanf("%d %d %d %d", &h_inicio, &m_inicio, &h_final, &m_final);

    if ((h_inicio < h_final) || (h_inicio == h_final && m_inicio < m_final)) {
        horas = h_final - h_inicio;
        minutos = m_final - m_inicio;
    } else if (h_inicio == h_final && m_inicio == m_final) {
        horas = 24;
        minutos = 0;
    } else {
        horas = (24 - h_inicio) + h_final;
        minutos = m_final - m_inicio;
    }
    if (minutos < 0) {
        minutos += 60;
        horas--;
    }
    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", horas, minutos);

    return 0;
}
